queries.sql

```diff
@@ -5,4 +5,79 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * from animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+-- Transaction 1: Set species to 'unspecified' for animals with names ending in 'mon'
+BEGIN;
+SELECT * FROM animals WHERE species = 'unspecified';
+SELECT * FROM animals;
+
+UPDATE animals
+SET species = 'unspecified'
+WHERE name LIKE '%mon';
+
+
+-- Rollback the transaction
+ROLLBACK;
+
+-- Verify that changes were rolled back
+SELECT * FROM animals;
+
+-- Transaction 2: Set species to 'digimon' for animals with names ending in 'mon', and 'pokemon' for animals with no species
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL OR species = '';
+
+-- Verify the changes
+SELECT * FROM animals WHERE species IN ('digimon', 'pokemon');
+
+-- Commit the transaction
+COMMIT;
+
+-- Verify that changes persist after commit
+SELECT * FROM animals WHERE species IN ('digimon', 'pokemon');
+
+-- Transaction 3: Delete all records and rollback
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+
+-- Verify if records still exist
+SELECT * FROM animals;
+
+-- Transaction 4: Delete records born after Jan 1st, 2022, and update weights
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT weight_update;
+UPDATE animals SET weight_kg = weight_kg * -1;
+SELECT * FROM animals;
+ROLLBACK TO weight_update;
+UPDATE animals SET weight_kg = weight_kg * -1;
```

Comment:
- [ ] Here, you are making the multiplication directly whereas, it should be under a condition of `WHERE weight_kg < 0`
![image](https://github.com/JoyTaribagshaw/vet-clinic/assets/86200356/d5048575-25e3-43ca-b0aa-33e51b17ab7a)

Reply:
Thank you so much for your review @Olamarx , I have made the required changes.

![changes](https://github.com/JoyTaribagshaw/vet-clinic/assets/111312788/5ecf05e2-114d-472f-8e41-25cf9f77ec4e)